source/Hardpoint.cpp

```diff
@@ -181,7 +182,20 @@ bool Hardpoint::CanAim() const
 // Check if this weapon is ready to fire.
 bool Hardpoint::IsReady() const
 {
-	return outfit && burstReload <= 0. && burstCount;
+	return outfit && burstReload <= 0. && burstCount && !IsBlind();
+}
+
+
+
+// Check if this weapon can't fire because of its blindspots.
+bool Hardpoint::IsBlind() const
+{
+	return any_of(baseAttributes.blindspots.begin(), baseAttributes.blindspots.end(),
+		[this](pair<Angle, Angle> blindspot)
+		{
+			return angle.Degrees() >= (blindspot.first + baseAngle).Degrees()
+				&& angle.Degrees() <= (blindspot.second + baseAngle).Degrees();
```

Comment:
Blindspots aren't normalized in any way, which makes this not work (probably). A blindspot in the range [170, 190] = [170, -170] cannot be used because the min angle would be greater than the max angle, and swapping them would swap what is blind with what is not.

I tried to test this, but couldn't get blindspots to work at all...